Jellyfin.Server/Migrations/Routines/MigrateActivityLogDb.cs

```diff
@@ -0,0 +1,101 @@
+#pragma warning disable CS1591
+
+using System;
+using System.IO;
+using Emby.Server.Implementations.Data;
+using Jellyfin.Data.Entities;
+using Microsoft.Extensions.Logging;
+using SQLitePCL.pretty;
+
+namespace Jellyfin.Server.Migrations.Routines
+{
+    public class MigrateActivityLogDb : IMigrationRoutine
+    {
+        public Guid Id => Guid.Parse("3793eb59-bc8c-456c-8b9f-bd5a62a42978");
+
+        public string Name => "MigrateActivityLogDatabase";
+
+        public void Perform(CoreAppHost host, ILogger logger)
+        {
+            var dataPath = host.ServerConfigurationManager.ApplicationPaths.DataPath;
+            using (var connection = SQLite3.Open(
+                Path.Combine(dataPath, "activitylog.db"),
+                ConnectionFlags.ReadOnly,
+                null))
+            {
+                logger.LogInformation("This may take a while, do not stop Jellyfin.");
+                using var dbContext = host.DatabaseProvider.GetConnection();
+
+                var queryResult = connection.Query("SELECT * FROM ActivityLog ORDER BY Id ASC");
+
+                // Make sure that the database is empty in case of failed migration due to power outages, etc.
+                dbContext.Database.EnsureDeleted();
+                dbContext.Database.EnsureCreated();
+
+                foreach (var entry in queryResult)
+                {
+                    var newEntry = new ActivityLog(
+                        entry[1].ToString(),
+                        entry[4].ToString(),
+                        entry[6].SQLiteType == SQLiteType.Null ? Guid.Empty : Guid.Parse(entry[6].ToString()),
+                        entry[7].ReadDateTime(),
+                        ParseLogLevel(entry[8].ToString()));
+
+                    if (entry[2].SQLiteType != SQLiteType.Null)
+                    {
+                        newEntry.Overview = entry[2].ToString();
+                    }
+
+                    if (entry[3].SQLiteType != SQLiteType.Null)
+                    {
+                        newEntry.ShortOverview = entry[3].ToString();
+                    }
+
+                    if (entry[5].SQLiteType != SQLiteType.Null)
+                    {
+                        newEntry.ItemId = entry[5].ToString();
+                    }
+
+                    dbContext.ActivityLogs.Add(newEntry);
+                    dbContext.SaveChanges();
```

Comment:
That would be what I would do normally, but this data needs to be inserted in order, and since EF Core provides no way to guarantee order during bulk inserts, this is what I did.

Reply:
Why do the records need to be inserted in-order? AFAIK insert order does not matter for SQL databases, since you always need to specify an `ORDER BY` clause to get sorted data

Reply:
There is code that references the ID (Insertion order) of the records